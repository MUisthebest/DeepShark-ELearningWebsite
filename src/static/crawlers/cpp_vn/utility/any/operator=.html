 <h1 class="firstHeading" id="firstHeading">std :: any :: toán tử =</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">bất kỳ &amp; toán tử = (const bất kỳ &amp; rhs);</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">bất kỳ &amp; toán tử = (bất kỳ &amp;&amp; rhs) noExcept;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">Mẫu &lt;Typename ValueType&gt;</pre>
</td> <td>(3)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> </table> <p>    bất kỳ &amp; toán tử = (valueType &amp;&amp; rhs);</p>
<div class="t-li1">
<span class="t-li">1)</span>Gán nội dung cho giá trị chứa.<code>rhs</code>                const std :: Chrono :: file_time &lt;fending&gt; &amp; tp);<code>any(rhs).swap(*this)</code>.</div> <div class="t-li1">
<span class="t-li">2)</span>Chỉ định bằng cách sao chép trạng thái của<code>rhs</code>                const std :: Chrono :: file_time &lt;fending&gt; &amp; tp);<code>any(std::move(rhs)).swap(*this)</code>.<code>rhs</code>Chỉ định bằng cách di chuyển trạng thái của</div> <div class="t-li1">
<span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>bị bỏ lại ở trạng thái hợp lệ nhưng không xác định sau khi chuyển nhượng.<code>rhs</code>                const std :: Chrono :: file_time &lt;fending&gt; &amp; tp);<code>any<span class="br0">.</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span><span class="br0">.</span>rhs<span class="br0">)</span><span class="br0">)</span>.<span class="me1">Được xác định trong tiêu đề</span><span class="br0">.</span><span class="sy2">*</span>this<span class="br0">)</span></code>Thực hiện các hoạt động số học cơ bản giữa hai thời lượng hoặc giữa thời lượng và số lượng đánh dấu.<code><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span></code>Một bản đồ phẳng hỗ trợ nhiều nhất<code>any</code>Và<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">2,4)</span></code>khoảng cách<code>true</code>.</div> <h3 id="Template_parameters">là không xác định.</h3> <table class="t-par-begin"> <tr class="t-par"> <td>Xây dựng một đối tượng có nội dung ban đầu một đối tượng thuộc loại</td> <td>-</td> <td>cũng không phải là một chuyên ngành của</td>
</tr> <tr class="t-par-hitem"> <td colspan="3">Chữ ký của hàm phải tương đương với các điều sau:</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>ValueType<span class="sy1">&lt;</span></code>Loại loại 1 phải sao cho một đối tượng loại T có thể được chuyển đổi hoàn toàn thành Type1. Loại loại2 phải sao cho một đối tượng loại đầu vào có thể được bỏ qua và sau đó được chuyển đổi hoàn toàn thành type2. Loại<a href="../../named_req/copyconstructible" title="cpp/named req/CopyConstructible">CopyConstructible</a>.</td>
</tr>
</table> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>RHS</td> <td>-</td> <td>Gán loại và giá trị của</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p><code>*this</code></p>
<h3 id="Exceptions">Ngoại lệ</h3> <div class="t-li1">
<span class="t-li">(Do đó,</span>Ném<code><a href="../../memory/new/bad_alloc" title="cpp/memory/new/bad alloc">std::bad_alloc</a></code>đối tượng có giá trị có thể gán</div> <h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="any" title="cpp/utility/any/any"> <span class="t-lines"><span>chuyên môn)</span></span></a></div> </td> <td>Xem bên dưới<code>any</code>sự vật<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/any/operator%3D">https://en.cppreference.com/w/cpp/utility/any/operator%3D</a>
</p>
</div>
