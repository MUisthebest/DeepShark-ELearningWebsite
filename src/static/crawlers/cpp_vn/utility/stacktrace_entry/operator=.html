 <h1 class="firstHeading" id="firstHeading">std :: stacktrace_entry :: toán tử =</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr stacktrace_entry &amp; toán tử = (const stacktrace_entry &amp; other) noexcept;</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Sao chép toán tử gán. Thay thế nội dung của<code>*this</code>với những người<code>other</code>.</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>khác</td> <td>-</td> <td>khác<code>stacktrace_entry</code>để gán từ</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p><code>*this</code>.</p>
<div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/stacktrace_entry/operator%3D">https://en.cppreference.com/w/cpp/utility/stacktrace_entry/operator%3D</a>
</p>
</div>
