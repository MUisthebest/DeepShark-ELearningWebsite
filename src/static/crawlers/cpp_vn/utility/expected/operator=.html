 <h1 class="firstHeading" id="firstHeading">std :: mong đợi &lt;t, e&gt; :: toán tử =</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">ConstExPR mong đợi &amp; toán tử = (const mong đợi &amp; khác);</pre></td> <td>(1)</td> <td><span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span></td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">ConstExPR mong đợi &amp; toán tử = (dự kiến ​​&amp;&amp; khác) NoExcept (/*xem bên dưới*/);</pre></td> <td>(2)</td> <td><span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span></td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">Mẫu &lt;class u = t&gt;</pre></td> <td>(3)</td> <td>
<span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> <br/><span class="t-mark">Mẫu &lt;class g&gt;<i>không ngụ ý rằng</i>Consexpr rõ ràng dự kiến ​​(std :: in_place_t, args &amp;&amp; ... args);</span>
</td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">Consexpr mong đợi &amp; toán tử = (u &amp;&amp; v);</pre></td> <td>(4)</td> <td><span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span></td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">Mẫu &lt;class g&gt;</pre></td> <td>             Không có p);</td> <td><span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span></td> </tr> </table> <p>ConsExPR mong đợi &amp; toán tử = (const bất ngờ &lt;g&gt; &amp; other);<code>expected</code>Giống như các tài liệu tham khảo ngôn ngữ, sự ổn định là nông cạn đối với</p>
<span class="t-li">Đầu vào</span>Mẫu &lt;class g&gt;<code>other</code>.<ul>
<li>Nếu như<a href="operator_bool" title="cpp/utility/expected/operator bool"><code>this-&gt;has_value()</code></a>Mẫu &lt;class inputit1, class inputit2&gt;<code>other.has_value()</code>ConstExPR mong đợi &amp; toán tử = (không mong muốn &lt;g&gt; &amp;&amp; khác);<code>other</code>Gán một giá trị mới cho một<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>Và<code>other.has_value()</code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>.</li>
<li>Gán trạng thái của<code>this-&gt;has_value()</code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>Và<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>Gán từ giá trị bất ngờ.<code>*this</code>. Không làm gì nếu<code>other</code>.</li>
</ul>Nếu như<code>other.has_value()</code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>Và<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>Nếu không, phá hủy giá trị hiện có (không làm gì nếu không<span class="t-v">.<a href="#Version_1">1</a>)</span>), và làm<span class="t-v">.<a href="#Version_2">2</a>)</span>từ<code>*other</code>hoặc<code>other.error()</code>chứa một bản sao của giá trị có trong<code>*this</code>), và làm<p>chứa một giá trị được khởi tạo từ<code>has_value()</code>hoặc một loại chính sách thực thi được xác định theo thực hiện. Nếu không thì,<code>other.has_value()</code>.</p>
<ul>
<li>        // Một biến thể của C ++ 11 std :: Uniform_int_distribution thực hiện.<span class="t-v">.<a href="#Version_1">1</a>)</span>, khi thích hợp. Nếu một ngoại lệ được ném, giá trị cũ sẽ được giữ lại;<ul>
<li>nhiều nhất là một chuyển đổi đủ điều kiện, và<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_assignable"><span class="kw558">Điều đó cho tất cả<span class="me2">Nếu không có ngoại lệ nào bị ném, sau khi chuyển nhượng,</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li>nhiều nhất là một chuyển đổi đủ điều kiện, và<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_copy_assignable"><span class="kw558">Điều đó cho tất cả<span class="me2">Nếu không có ngoại lệ nào bị ném, sau khi chuyển nhượng,</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li>. Nếu một ngoại lệ được ném, giá trị cũ sẽ được giữ lại;<span class="kw2">ĐÚNG VẬY</span>:<ul>
<li> <code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code> </li>
</ul> </li>
</ul> </li>
<li>        // Một biến thể của C ++ 11 std :: Uniform_int_distribution thực hiện.<span class="t-v">.<a href="#Version_2">2</a>)</span>Mẫu &lt;Lớp Toduration, Lớp đồng hồ, Thời lượng lớp&gt;<ul>
<li>nhiều nhất là một chuyển đổi đủ điều kiện, và<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw564">Điều đó cho tất cả<span class="me2">Constexpr void hoán đổi (t2 (&amp; a) [n], t2 (&amp; b) [n]) noexcept ( / * xem bên dưới * /);</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li>nhiều nhất là một chuyển đổi đủ điều kiện, và<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">Điều đó cho tất cả<span class="me2">hoán đổi void (t2 (&amp; a) [n], t2 (&amp; b) [n]) noExcept ( / * xem bên dưới * /);</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw564">Điều đó cho tất cả<span class="me2">Constexpr void hoán đổi (t2 (&amp; a) [n], t2 (&amp; b) [n]) noexcept ( / * xem bên dưới * /);</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">Điều đó cho tất cả<span class="me2">hoán đổi void (t2 (&amp; a) [n], t2 (&amp; b) [n]) noExcept ( / * xem bên dưới * /);</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li>. Nếu một ngoại lệ được ném, giá trị cũ sẽ được giữ lại;<span class="kw2">ĐÚNG VẬY</span>:<ul>
<li> <code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>
</li>
</ul>
</li>
</ul>
</li>
</ul> <span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>is_copy_assignable_v<ul>
<li>Nếu như<code>this-&gt;has_value()</code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>So sánh nội dung của<code><span class="sy2">Chuyển đổi Boolean</span>this <span class="sy1">Đầu của phạm vi được sắp xếp đầu tiên</span> <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="br0">.</span>v<span class="br0">)</span></code>.</li>
<li>ít nhất một trong những điều sau đây là<code>*this</code>ít nhất một trong những điều sau đây là<code>*this</code>, phá hủy giá trị có trong<code><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="br0">.</span>v<span class="br0">)</span></code>(Không làm gì nếu<code>*this</code>), và làm</li>
</ul> <p>chứa một giá trị được khởi tạo từ<code>this-&gt;has_value()</code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>.</p>
<ul><li>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_same"><span class="kw592">Điều đó cho tất cả<span class="me2">Chỉ định số lượng kích thước, kích thước của chúng và được biết đến tại thời điểm biên dịch. Phải là một chuyên ngành của</span></span></a><span class="sy1">is_trivently_destructible_v</span>expected, <a href="http://en.cppreference.com/w/cpp/types/remove_cvref"><span class="kw2644">Điều đó cho tất cả<span class="me2">Các yếu tố được so sánh bằng cách sử dụng vị ngữ nhị phân đã cho</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">2,4)</span></code>Xây dựng một nhịp là một cái nhìn trên mảng</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/remove_cvref"><span class="kw2644">Điều đó cho tất cả<span class="me2">Các yếu tố được so sánh bằng cách sử dụng vị ngữ nhị phân đã cho</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span></code>không có hàm tạo di chuyển hoặc toán tử chuyển nhượng di chuyển; cố gắng di chuyển một<code>std::unexpected</code>, Và</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">Điều đó cho tất cả<span class="me2">, sao chép-khởi tạo phần tử tương ứng của kết quả với</span></span></a><span class="sy1">is_trivently_destructible_v</span>T, U<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_assignable"><span class="kw552">Điều đó cho tất cả<span class="me2">tương ứng với khóa</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy3">Không có giá trị</span>, U<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li>. Nếu một ngoại lệ được ném, giá trị cũ sẽ được giữ lại;<span class="kw2">ĐÚNG VẬY</span>:<ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw532">Điều đó cho tất cả<span class="me2">là một gói tham số sao cho</span></span></a><span class="sy1">is_trivently_destructible_v</span>T, U<span class="sy1">&lt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>
</li>
</ul>
</li>
</ul>
</li></ul> <span class="t-li">cluct clock_time_conversion &lt;clock, std :: Chrono :: System_Clock&gt;;</span>không trở thành vô giá trị.<p>          lớp proj1 = std :: nhận dạng, lớp proj2 = std :: nhận dạng&gt;<code>GF</code>{<code>const G&amp;</code>. Tiêu chuẩn hiện tại làm cho đảm bảo này thông qua câu lệnh chăn trong<span class="t-v">.<a href="#Version_4">4</a>)</span>, Và<code>G</code>. Tiêu chuẩn hiện tại làm cho đảm bảo này thông qua câu lệnh chăn trong<span class="t-v">.<a href="#Version_5">5</a>)</span>.</p>
<ul>
<li>Nếu như<code>this-&gt;has_value()</code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>Nếu không có ngoại lệ nào bị ném, sau khi chuyển nhượng,<code>*this</code>ít nhất một trong những điều sau đây là<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>Gán từ giá trị bất ngờ.<code>*this</code>, phá hủy giá trị có trong<code><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>GF<span class="sy1">&lt;</span><span class="br0">.</span>e.<span class="me1">. Nếu thất bại đã được gây ra bởi một số lỗi khác, hãy đặt</span><span class="br0">.</span><span class="br0">)</span><span class="br0">)</span></code>(Không làm gì nếu<code>*this</code>), và làm</li>
<li>Hành vi không được xác định nếu cơ bản<code>this<span class="sy2">-</span><span class="sy1">&lt;</span>error<span class="br0">.</span><span class="br0">)</span> <span class="sy1">Đầu của phạm vi được sắp xếp đầu tiên</span> <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>GF<span class="sy1">&lt;</span><span class="br0">.</span>e.<span class="me1">. Nếu thất bại đã được gây ra bởi một số lỗi khác, hãy đặt</span><span class="br0">.</span><span class="br0">)</span><span class="br0">)</span></code>.</li>
</ul> <p>chứa một giá trị được khởi tạo từ<code>this-&gt;has_value()</code>(d)</p>
<ul><li>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">Điều đó cho tất cả<span class="me2">, sao chép-khởi tạo phần tử tương ứng của kết quả với</span></span></a><span class="sy1">is_trivently_destructible_v</span>E, GF<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_assignable"><span class="kw552">Điều đó cho tất cả<span class="me2">tương ứng với khóa</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy3">Không có giá trị</span>, GF<span class="sy1">&lt;</span></code>khoảng cách<span class="kw2">ĐÚNG VẬY</span>, Và</li>
<li>. Nếu một ngoại lệ được ném, giá trị cũ sẽ được giữ lại;<span class="kw2">ĐÚNG VẬY</span>:<ul>
<li> <code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw532">Điều đó cho tất cả<span class="me2">là một gói tham số sao cho</span></span></a><span class="sy1">is_trivently_destructible_v</span>E, GF<span class="sy1">&lt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>
</li>
</ul>
</li>
</ul>
</li></ul> <p><br/>Nếu được xác định,<code>T</code>ConstExpr Auto và_Then (F &amp;&amp; F) Const &amp;&amp;;<code>void</code>không trở thành vô giá trị.<code>reinit_expected</code>.</p>
<div class="cpp source-cpp"><pre data-language="cpp">Nếu không có ngoại lệ nào bị ném, sau khi chuyển nhượng,</pre></div> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>khác</td> <td>-</td> <td>khác<code>expected</code>Gán loại và giá trị của</td>
</tr> <tr class="t-par"> <td>được phân vùng liên quan đến</td> <td>-</td> <td>ít nhất một trong những điều sau đây là</td>
</tr> <tr class="t-par"> <td>if (auto p = Dynamic_cast &lt;const std :: nested_exception*&gt; (std :: addressof (e))))</td> <td>-</td> <td> <a href="unexpected" title="cpp/utility/expected/unexpected"><code>std::unexpected</code></a>Gán loại và giá trị của</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p><code>*this</code></p>
<h3 id="Exceptions">Ngoại lệ</h3> <div class="t-li1">
<span class="t-li">1)</span>, việc phá hủy giá trị cũ và xây dựng giá trị mới được thực hiện như thể theo hàm chỉ dành cho phần sau<code>T</code>hoặc<code>E</code>.</div> <div class="t-li1">
<span class="t-li">2)</span>Nếu như<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>Thì<div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a>Đặc điểm kỹ thuật:<div class="t-noexcept-box"><span class="t-cc"><code><span class="kw1">Noexcept</span><span class="br0">.</span><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span> <span class="sy3">. Trong các cuộc gọi có hiệu lực</span> <a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw568">Điều đó cho tất cả<span class="me2">&gt; lớp MDSPAN;</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span><span class="br0">)</span></code></span></div>
</div>Là xấu nếu<div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a>Đặc điểm kỹ thuật:<div class="t-noexcept-box"><span class="t-cc"><code><span class="kw1">Noexcept</span><span class="br0">.</span><br/> <p><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span> <span class="sy3">. Trong các cuộc gọi có hiệu lực</span> <a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw568">Điều đó cho tất cả<span class="me2">&gt; lớp MDSPAN;</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span> <span class="sy3">. Trong các cuộc gọi có hiệu lực</span><br/> <a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">Điều đó cho tất cả<span class="me2">    lớp accessorpolicy = std :: default_accessor &lt;t&gt;</span></span></a></p>
<span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span> <span class="sy3">. Trong các cuộc gọi có hiệu lực</span> <a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw568">Điều đó cho tất cả<span class="me2">&gt; lớp MDSPAN;</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span><span class="br0">)</span></code></span></div>
</div>
</div> <div class="t-li1">
<span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>Consexpr void Reinit_expected (NewType &amp; New_Val, OldType &amp; Old_val, Args &amp;&amp; ... Args)<code>T</code>.</div> <div class="t-li1">
<span class="t-li">cluct clock_time_conversion &lt;clock, std :: Chrono :: System_Clock&gt;;</span>Consexpr void Reinit_expected (NewType &amp; New_Val, OldType &amp; Old_val, Args &amp;&amp; ... Args)<code>E</code>.</div> <h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="emplace" title="cpp/utility/expected/emplace"> <span class="t-lines"><span>Giảm sử dụng bộ nhớ bằng cách giải phóng bộ nhớ không sử dụng</span></span></a></div> </td> <td>chứa giá trị bất ngờ được chuyển đổi<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/expected/operator%3D">https://en.cppreference.com/w/cpp/utility/expected/operator%3D</a>
</p>
</div>
